src/Storages/StorageMemory.cpp

```diff
@@ -31,7 +31,7 @@ class MemorySource : public SourceWithProgress
         : SourceWithProgress(metadata_snapshot->getSampleBlockForColumns(column_names_, storage.getVirtuals(), storage.getStorageID()))
         , column_names(std::move(column_names_))
         , begin(begin_)
-        , end(end_)
+        , end(end_) /// [begin, end]
```

Comment:
If it's included, name them first, last.

Reply:
Need a comment about the semantics of the `end` list iterator in presense of insertions.